src/app/pages/system/tunable/tunable-form/tunable-form.component.ts

```diff
@@ -1,67 +1,88 @@
 import { ApplicationRef, Component, Injector } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { helptext_system_tunable as helptext } from 'app/helptext/system/tunable';
-import { RestService, WebSocketService } from '../../../../services/';
+import { WebSocketService } from '../../../../services/';
 import { FieldConfig } from '../../../common/entity/entity-form/models/field-config.interface';
+import { FieldSet } from 'app/pages/common/entity/entity-form/models/fieldset.interface';
 
 @Component({
   selector: 'system-tunable-edit',
   template: `<entity-form [conf]="this"></entity-form>`
 })
 export class TunableFormComponent {
-
-  protected resource_name: string = 'system/tunable';
+  protected queryCall = 'tunable.query';
+  protected queryKey = 'id';
+  protected editCall = 'tunable.update';
+  protected addCall = 'tunable.create';
+  protected pk: any;
+            
   protected route_success: string[] = ['system', 'tunable'];
   protected isEntity: boolean = true;
 
-  protected fieldConfig: FieldConfig[] = [{
-      type: 'input',
-      name: 'tun_var',
-      placeholder: helptext.var.placeholder,
-      tooltip: helptext.var.tooltip,
-      required: true,
-      validation : helptext.var.validation
-    },
-    {
-      type: 'textarea',
-      name: 'tun_value',
-      placeholder: helptext.value.placeholder,
-      tooltip: helptext.value.tooltip,
-      required: true,
-      validation : helptext.value.validation
-    },
+  protected fieldConfig: FieldConfig[] = [];
+  protected fieldSets: FieldSet[] = [
     {
-      type: 'select',
-      name: 'tun_type',
-      placeholder: helptext.type.placeholder,
-      tooltip: helptext.type.tooltip,
-      options: [
-        { label: 'loader', value: 'loader' },
-        { label: 'rc.conf', value: 'rc' },
-        { label: 'sysctl', value: 'sysctl' },
+      name: helptext.metadata.fieldsets[0],
+      class:'add-cron',
+      label:true,
+      width:'300px',
+      config:[
+        {
+          type: 'input',
+          name: 'var',
+          placeholder: helptext.var.placeholder,
+          tooltip: helptext.var.tooltip,
+          required: true,
+          validation : helptext.var.validation
+        },
+        {
+          type: 'textarea',
+          name: 'value',
+          placeholder: helptext.value.placeholder,
+          tooltip: helptext.value.tooltip,
+          required: true,
+          validation : helptext.value.validation
+        },
+        {
+          type: 'select',
+          name: 'type',
```

Comment:
type appears to be a required variable, if you try to save without setting type it gives an unhelpful error